### PR TITLE
Bumping versions to unify them

### LINKED ISSUE #5209: Unify versions between Editors/AppDesigner and Managed/Managed.VS
These dlls have different versions for no actual reason. Just unify so that we have one version (choose the higher version to avoid binding redirect breaks):

https://github.com/dotnet/project-system/blob/master/build/Versions.props#L3-L6

### PR DESCRIPTION
**Pull request by dmonroym**
Fixes #5209

### CODE REVIEW

File: build/Versions.props [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <VisualStudioVersion>16.3.0</VisualStudioVersion>
-    <ProjectSystemVersion>3.3.0</ProjectSystemVersion>
+    <ProjectSystemVersion>16.3.0</ProjectSystemVersion>
```
Comment (tmeschter) [inline build/Versions.props]: If they are going to have the same value, can't we just get rid of one entirely?

File: build/Versions.props [hunk 2]
```diff
@@ -1,9 +1,7 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <VisualStudioVersion>16.3.0</VisualStudioVersion>
-    <ProjectSystemVersion>3.3.0</ProjectSystemVersion>
-    <VersionBase Condition="'$(UseVisualStudioVersion)' == 'true'">$(VisualStudioVersion)</VersionBase>
```
Comment (davkean) [inline build/Versions.props]: Can we remove this UseVisualStudioVersion throughout the code base?

File: build/Versions.props (hunk 1 above)
Comment (davkean) [reply build/Versions.props]: You'll need to update usage throughout the tree, for example: GenerateDependentAssemblyVersionFile.proj.

Comment (dmonroym) [reply build/Versions.props]: @davkean What is GenerateDependentAssemblyVersionFile.proj? I can't seem to find that anywhere

Comment (davkean) [reply build/Versions.props]: https://github.com/dotnet/project-system/blob/master/build/Scripts/GenerateDependentAssemblyVersionFile.proj

Comment (davkean) [reply build/Versions.props]: What I do when I change/remove a variable is the following:

```
project-system>findstr /sip /c:"VisualStudioVersion" *
```

Comment (dmonroym) [reply build/Versions.props]: Ah that's very useful. Somehow when I do a search for this file it's not indexed - seems to be true for most files under the /build directory.

Comment (davkean) [reply build/Versions.props]: Another thing to be aware of, is that you also need to search the toolset under:

```
C:\Users\[user]\.nuget\packages\roslyntools.repotoolset
```

To see if it consumes it.

**Comment by davkean** [comment]
Note these files moved in https://github.com/dotnet/project-system/pull/5277.

**Comment by davkean** [comment]
I think you'll want to rebase this change.

File: build/Versions.props (hunk 1 above)
Comment (dmonroym) [reply build/Versions.props]: roslyntools.repotoolset doesn't consume it

File: build/Versions.props (hunk 2 above)
Comment (dmonroym) [reply build/Versions.props]: `UseVisualStudioVersion` is no longer used

File: src/setup/VisualStudioEditorsSetup/Directory.Build.props [hunk 3]
```diff
@@ -1,11 +1,8 @@
 <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
```
Comment (davkean) [inline src/setup/VisualStudioEditorsSetup/Directory.Build.props]: These empty Build.props files can be deleted now and the projects will automatically pick up the parent.

**Pull request merged**